dist/tools/vera++/scripts/rules/T003.tcl

```diff
@@ -38,7 +47,10 @@ foreach f [getSourceFileNames] {
             }
             set state "other"
         } else {
-            if [isKeyword $tokenName] {
+            # Check if it's a C++ file
+            regexp {\.[ch]pp} $f m
+            # C keywords are valid in C++ but not the other way around
+            if {[isCKeyword $tokenName] || ([info exists m] && [isCPPKeyword $tokenName])} {
```

Comment:
But I guess this `||` has the same effect?

Reply:
yes, that's what I did. It's just the semantics of the isCPPkeyword function :)